testing/oauth_server.ts

```diff
@@ -164,3 +199,38 @@ class OAuthServerContainer {
     this._server?.close();
   }
 }
+
+export async function performOAuthClientCredentialsServerFlow({
+  clientId,
+  clientSecret,
+  authDef,
+  scopes,
+  afterTokenExchange,
+}: {
+  clientId: string;
+  clientSecret: string;
+  authDef: OAuth2ClientCredentialsAuthentication;
+  scopes?: string[];
+  afterTokenExchange?: (params: AfterTokenOAuthClientCredentialsExchangeParams) => void;
```

Comment:
This is very awkward in that we have a callback and it returns the credentials. The callback is necessary so we can properly stub this function in the tests. The regular returns value is for when this gets called in an async function during the refresh. We could just use the callback in both places, though it may cause the code in the refresh case to look more awkward

Reply:
Hmm I don't think the real reason is because of tests, I think it's because in the regular OAuth2 case you're waiting for user input. You launch a server that idles while the user goes to their browser to go through the approval flow on the web, and only when they complete that is there sufficient info to run the token exchange, so by necessity it's an interactive flow.

Client credentials isn't interactive so I don't think we should be modeling it the same way. The code to make the exchange should just be a normal async function that gets awaited.

Reply:
I tried to model it like that initially. However `setupAuth`, which will eventually call this function, wasn't async, so I thought we needed either a `.then` or a callback to set the credentials afterwards. `.then` didn't work since the promise would let go of execution and `executeFetch` here would get the incomplete credentials:
https://github.com/coda/packs-sdk/pull/2798/files#diff-56a0e2bb7f87e922174aa0681e794985f029bc4718805e6434f006248eac5509R1343

 `setupAuth` is only called from the `setupAuthFromModule` which is async, so I'm wonder if we can just make it async as well

Reply:
I've switched `setupAuth` to be async and removed the callback, and it looks to be working fine